proto/proxy/destination/destination.proto

```diff
@@ -42,3 +68,7 @@ message WeightedAddr {
   common.TcpAddress addr = 1;
   uint32 weight = 3;
 }
+
+message NoEndpoints {
```

Comment:
In terms of naming, I'm inclined to call this message `ServiceStatus` or something similar. I realize that the other two message types return endpoints and this one does not, but I think picking a name that's more reflective of what it does return would be helpful.

Reply:
+1, prior to seeing this PR i was reviewing #568 and was confused by messages like this:
```bash
$ ./bin/go-run controller/script/destination-client -path books.default.svc.cluster.local:7002
NoEndpoints:
- exists:false

...

NoEndpoints:
- exists:true
```

Reply:
I'm wary of talking about _Services_ here -- nothing in this file talks about services, and i don't think that this response should be considered to only reflect kuberentes `Service` status...

The most consistent naming would probably be `NoAddrs`...

Reply:
In my mind `NoAddrs` suffers from the same issue as `NoEndpoints` -- it's naming the message in terms of what it doesn't do, whereas I would prefer to have some indication of what it does do.

How about something like:

```
message StatusUpdate {
  bool entity_exists = 1;
}
```

Reply:
`NoEndpoints { }` means "there are no endpoints for the service; if you previously found some, then forget about them". If we rename this to `ServiceStatus` or `StatusUpdate` then we lose that very important aspect of the name. The Rust code (before this PR) already has a bug where it fails to reset the set of endpoints when it reconnects.

As far as "service" vs. "entity" or other terms, the destination service is already documented as the "service discovery API" so I think "service" is the most appropriate term. I admit I chose "endpoints" based on the Kubernetes terminology.

Perhaps we could have the naming be `Reset { bool service_exists = 1 }` to avoid the "endpoints" term and lea

Reply:
i like Reset. it makes sense in the context of add/remove events

Reply:
Yep, `Reset { bool service_exists = 1 }` works for me.